### ISSUE TITLE
llava1.6_llama3.1_8B使用报错error occurs in lazy tokenizer: pixel values

### ISSUE THREAD

**Issue by JHL328** [modelscope/ms-swift#2409]
1. 已经git pull了最新的code
2. 在使用llava1.6_llama3.1_8B的时候报错
{'loss': 1.4909668, 'grad_norm': 11.20178784, 'learning_rate': 0.0, 'memory(GiB)': 34.69, 'train_speed(iter/s)': 0.03356, 'ep
och': 0.0, 'global_step/max_steps': '1/604', 'percentage': '0.17%', 'elapsed_time': '24s', 'remaining_time': '4h 9m 38s'}    
Train:   0%|▏                                                                              | 1/604 [00:24<4:09:31, 24.83s/it]
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'                                                                  
Train:   0%|▎                                                                              | 2/604 [00:35<2:44:51, 16.43s/it]
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'
[ERROR:swift] Error occurs in lazy tokenize: 'pixel_values'
使用脚本如下，脚本应该没问题，之前ft了其他模型
NPROC_PER_NODE=8 CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 swift sft \
    --model_type llava1_6-llama3_1-8b \
    --model_id_or_path .cache/modelscope/hub/swift/llava-llama3___1-8b \
    --dataset updated_html.jsonl \
    --sft_type full \
    --output_dir output \
    --max_length 16384 \
    --ddp_backend nccl \
    --freeze_vit True \
    --deepspeed default-zero3 \
3. report a bug: 在ms-swift提供的llava1.6_llama3.1_8B-instruct里面，连接早已经失效，一直没有更新，请尽快更新
![image](https://github.com/user-attachments/assets/c2e46962-8819-46ff-88e5-5db24aec2772)
![image](https://github.com/user-attachments/assets/05b2951c-56d7-4c33-9b8f-d196a6cb4302)

**Comment by yingdachen** [comment]
model link fixed. 

sft issue @Jintao-Huang

**Comment by DaozeZhang** [comment]
看上去是你的数据在`swift/llm/utils/utils.py`的257行执行`res = self.encode_func(data)`时出了问题，大概率是这里面尝试访问了`pixel_values`这个key，但这个key不存在。

用其他图像数据集（例如coco-en-2-mini）是可以跑通这一步的，所以请你看一下自己的自定义数据集格式是否有问题。可以单步进入试试。

**Comment by JHL328** [comment]
> 看上去是你的数据在`swift/llm/utils/utils.py`的257行执行`res = self.encode_func(data)`时出了问题，大概率是这里面尝试访问了`pixel_values`这个key，但这个key不存在。
> 
> 用其他图像数据集（例如coco-en-2-mini）是可以跑通这一步的，所以请你看一下自己的自定义数据集格式是否有问题。可以单步进入试试。

我也尝试了用其他数据集，coco-en-2-mini，确实在运行的时候是没有这个问题的。
下面是我定义的数据格式，是符合格式要求的（ft 过其他的模型没有出现pixel values报错，一样的数据格式）
{"query": "<image>Look at the design in the image and generate HTML code according to material design principles.,generate corresponding html code", 
"response": "<!DOCTYPE htm。。。。“
"images": "/image/example.png"}

**Comment by JHL328** [comment]
已经解决，在/swift/llm/utils的template.py的里面有继承的错误，需要把上面父类的一部分复制过来
![image](https://github.com/user-attachments/assets/938057f3-c1c9-4616-851b-692a7d448991)
